ruby/lib/gherkin3/token_scanner.rb

```diff
@@ -11,15 +13,23 @@ module Gherkin3
   # to look for Gherkin keywords for the associated language. The keywords are defined 
   # in gherkin-languages.json.
   class TokenScanner
+    protected
+
+    attr_reader :io
```

Comment:
Please remove this. Not needed when `Parser.parse` looks at the type of the first arg to decide whether it's already a scanner or if it needs to wrap a scanner around a string or io.